content/030_base/035_gitrepo/010-create-repo.md

```diff
@@ -0,0 +1,463 @@
+}
+output "codecommit_key_name" {
+  description = "Secret name that holds the SSH key for accessing CodeCommit"
+  value       = aws_secretsmanager_secret.codecommit_key.name
+}
+
+EOF
+```
+### 5. Provision CodeCommit repositories
+
+```bash
+cd ~/environment/codecommit
+terraform init
+terraform apply --auto-approve
+```
+
+## Populate git repositories
+
+The CodeCommit repositories will be populated with starter files first. These starter files will provide a foundation for the workshop. In the following workshop chapters, we will build on top of these starter files.
+### 1. Set environment variables
+
+```bash
+export ACCOUNT_ID=$(aws sts get-caller-identity --output text --query Account)
+export AWS_DEFAULT_REGION=$(curl -s 169.254.169.254/latest/dynamic/instance-identity/document | jq -r '.region')
+export WORKING_DIR="$HOME/environment"
+export SOURCE_DIR="$WORKING_DIR/source/assets"
+export SCRIPT_DIR="$SOURCE_DIR/scripts"
+export GITOPS_DIR="$WORKING_DIR/gitops-repos"
+
+
+echo "export ACCOUNT_ID=${ACCOUNT_ID}" | tee -a ~/.bash_profile
+echo "export AWS_DEFAULT_REGION=${AWS_DEFAULT_REGION}" | tee -a ~/.bash_profile
+echo "export WORKING_DIR=${WORKING_DIR}" | tee -a ~/.bash_profile
+echo "export SOURCE_DIR=${SOURCE_DIR}" | tee -a ~/.bash_profile
+echo "export SCRIPT_DIR=${SCRIPT_DIR}" | tee -a ~/.bash_profile
+echo "export GITOPS_DIR=${GITOPS_DIR}" | tee -a ~/.bash_profile
+source ~/.bash_profile
+
+```
+
+
+
+### 2. Clone starter files
+
+Clone webstore workload , starter files for platform and cleanup scripts.
+
+![Clone Repository](/static/images/clone_starterfiles.png)
+
+```bash
+cd $WORKING_DIR
+git clone --depth 1 --no-checkout https://github.com/aws-samples/eks-blueprints-for-terraform-workshop source
+cd $WORKSHOP_DIR
+git sparse-checkout set assets 
+git checkout
+cd $WORKING_DIR
+
+```
+::::expand{header="What is in my cloned repo?"}
+This repository contains resources for managing Kubernetes clusters in the **assets** directory. It includes Kubernetes YAML files for deploying workloads, ApplicationSets, and configuration values for addons, namespaces, and projects.
+
+Asset Folder:![Asset Folders](/static/images/asset-github-folders.png)
+
+Platform Folder:![Platform Folders](/static/images/platform-github-folders.png)
+
+Webstore Workload Folder:![Workload Folders](/static/images/workload-github-folders.png)
+::::
+
+### 3. Populate codecommit gitops-platform repository
+
+Copy platform starter files to "gitops-repos" folder from the cloned repository.
+
+![Local Platform](/static/images/local_platform.png)
+
+Push "gitops-repos" platform folder to codecommit "gitops-platform" repository
+
+![CodeCommit Platform](/static/images/codecommit_platform.png)
+
+
+```bash
+mkdir -p ${GITOPS_DIR}
+gitops_platform_url=https://git-codecommit.${AWS_DEFAULT_REGION}.amazonaws.com/v1/repos/gitops-platform
+# populate platform repository
+git clone ${gitops_platform_url} ${GITOPS_DIR}/platform
```

Comment:
This is asking me login and password

Reply:
Cloning into '/home/ec2-user/environment/gitops-repos/platform'...
Username for 'https://git-codecommit.us-east-2.amazonaws.com/v1/repos/gitops-platform':